src/features/github-download-button.js

```diff
@@ -1,3 +1,129 @@
+import React from 'dom-chef';
+import onDomReady from 'dom-loaded';
+
+import { isGitHub, isRepoRoot } from '../libs/page-detect';
+import { getRepoDetails } from '../libs/utils';
+import { clippy } from '../libs/icons';
+
 import './github-download-button.css';
 
-console.log('🚧 jsDelivr Extension 🚧');
+async function init () {
+	await onDomReady;
+
+	if (isGitHub()) {
+		if (isRepoRoot()) {
+			try {
+				if (document.querySelector('.jsd-get-repo-select-menu')) {
+					return;
+				}
+
+				const { ownerName, repoName, treeName } = getRepoDetails();
+
+				const rawPackageUrl = `https://raw.githubusercontent.com/${ownerName}/${repoName}/${treeName}/package.json`;
+				const rawPackageResponse = await fetch(rawPackageUrl);
+				const { name, version } = await rawPackageResponse.json();
+
+				if (!name && !version) {
```

Comment:
`||` ?

Reply:
We need both `name` and `version` for our API to get default file name from.

Reply:
Sure, so if either is missing it should go to the `return`

Reply:
Yes.

Reply:
But right now both need to be missing 😄

Reply:
Yes. Regarding this point, I found 1 use-case that I missed.
[Explained here](https://github.com/jsdelivr/extension-chrome/issues/1#issuecomment-537517324).